### PR TITLE
[ API ] capi-nntrainer packaging

### PR DESCRIPTION
**Pull request by jijoongmoon**
This PR includes:
  . rpm packaging of capi-nntrainer for tzien

related #265 

**Self evaluation:** 
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: packaging/nntrainer.spec [hunk 1]
```diff
@@ -79,6 +90,35 @@ Summary:	NNTrainer UnitTest Coverage Analysis Result
...
 %endif
 
+%if %{with tizen}
+%package -n capi-nntrainer
+Summary:         Tizen Native API for NNTrainer
+Group:           Multimedia/Framework
+Requires:        %{name}=%{version}-%{release}
+%description -n capi-nntrainer
+Tizen Native API wrapper for NNTrainer.
+You can trian neural networks efficiently.
```
Comment (kparichay) [inline packaging/nntrainer.spec:100]: `s/trian/train/g`

File: packaging/nntrainer.spec [hunk 2]
```diff
@@ -59,6 +62,12 @@ Requires:	openblas-devel
 Development pacage for custom nntrainer developers.
```
Comment (zhoonit) [inline packaging/nntrainer.spec:62]: Not in this PR though...
```suggestion
Development package for custom nntrainer developers.
```

File: packaging/nntrainer.spec [hunk 3]
```diff
@@ -196,15 +240,32 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
...
+%files -n capi-nntrainer
+%manifest capi-nntrainer.manifest
+%license LICENSE
+%{_libdir}/libcapi-nntrainer.so
+
+%files -n capi-nntrainer-devel
+%{_includedir}/nntrainer/nntrainer.h
+%{_libdir}/pkgconfig/capi-nntrainer.pc
+
+%files -n capi-nntrainer-devel-static
```
Comment (zhoonit) [inline packaging/nntrainer.spec:264]:
```suggestion
%files -n capi-nntrainer-devel-static
%{_includedir}/nntrainer/nntrainer.h 
```

File: packaging/nntrainer.spec [hunk 4]
```diff
@@ -196,15 +240,32 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
...
+
+%if %{with tizen}
+%files -n capi-nntrainer
+%manifest capi-nntrainer.manifest
+%license LICENSE
+%{_libdir}/libcapi-nntrainer.so
+
+%files -n capi-nntrainer-devel
+%{_includedir}/nntrainer/nntrainer.h
+%{_libdir}/pkgconfig/capi-nntrainer.pc
```
Comment (zhoonit) [inline packaging/nntrainer.spec:262]: Is this okay not to add *.so file here?

Comment (jijoongmoon) [reply packaging/nntrainer.spec:262]: i requres cpi-nntrainer which has .so. no need to include *.so.

Comment (zhoonit) [reply packaging/nntrainer.spec:262]: Got it. Thanks for the explanation :)

**Comment by zhoonit** [comment]
LGTM after @kparichay issue has been resolved.

File: packaging/nntrainer.spec [hunk 5]
```diff
@@ -79,6 +90,35 @@ Summary:	NNTrainer UnitTest Coverage Analysis Result
...
+Group:           Multimedia/Framework
+Requires:        capi-nntrainer = %{version}-%{release}
+Requires:        capi-ml-common-devel
+%description -n capi-nntrainer-devel
+Developmental kit for Tizen Native NNTrainer API.
+
+%package -n capi-nntrainer-devel-static
+Summary:         Static library for Tizen Native API
+Group:           Multimedia/Framework
+Requires:        capi-nntrainer-devel=%{versin}-%{release}
```
Comment (myungjoo) [inline packaging/nntrainer.spec:116]: - Typo (versin)
- Style: Add spaces around = and version's -

File: packaging/nntrainer.spec (hunk 3 above)
Comment (myungjoo) [reply packaging/nntrainer.spec:264]: As long as capi-nntrainer-devel-static requires capi-nntrainer-devel, it does not need to have it and having a same file in two sub-packakges is dangerous (I personally prohibit it)

File: packaging/nntrainer.spec (hunk 1 above)
Comment (jijoongmoon) [reply packaging/nntrainer.spec]: fixed. thanks.

File: packaging/nntrainer.spec (hunk 5 above)
Comment (jijoongmoon) [reply packaging/nntrainer.spec]: fixed. Thanks.

File: packaging/nntrainer.spec (hunk 3 above)
Comment (zhoonit) [reply packaging/nntrainer.spec:264]: I see. Thanks for the explanation :)

File: api/capi/capi-nntrainer.pc.in [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+prefix=@PREFIX@
+exec_prefix=@PREFIX@
+libdir=@LIB_INSTALL_DIR@
+includedir=@INCLUDE_INSTALL_DIR@
+
+Name: tizen-api-nntrainer
+Description: NNTrainer API for Tizen
+Version: @VERSION@
+Requires:
+Libs: -L${libdir} -lcapi-nntrainer
```
Comment (kparichay) [inline api/capi/capi-nntrainer.pc.in:11]: -lnntrainer??

File: api/capi/capi-nntrainer.pc.in [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Package Information for pkg-config
```
Comment (kparichay) [inline api/capi/capi-nntrainer.pc.in:1]: I am not sure, but maybe this file can be better placed in packaging?

**Pull request merged**